pandas/io/formats/format.py

```diff
@@ -227,6 +238,7 @@ def _chk_truncate(self):
         truncate_v = max_rows and (len(self.series) > max_rows)
         series = self.series
         if truncate_v:
+            max_rows = cast(int, max_rows)
```

Comment:
Is this required? Generally we try to avoid cast and I'm surprised MyPy can't narrow the type down here from None. 

If it's fully required does making line 238 say `max_rows is not None and ...` get around that?

Reply:
> If it's fully required does making line 238 say `max_rows is not None and ...` get around that?

no. gives `pandas\io\formats\format.py:249: error: Unsupported operand types for // ("None" and "int")` and there _might_ be risk in implementing this differently anyway?

> Generally we try to avoid cast

well I would have agreed with you when I started this, but after this PR, I think that adding type annotations should be just that and any refactoring or code cleanup should be deferred to a follow on PR.

> I'm surprised MyPy can't narrow the type down here from None.

Agreed. hence the numerous iterations to try and avoid cast originally.

Reply:
Hmm OK yea seems like a bug / gap with mypy inference then. Can always come back to clean up